mm2src/coins/utxo/bch.rs

```diff
@@ -184,20 +186,43 @@ impl BchCoin {
     pub fn bchd_urls(&self) -> &[String] { &self.bchd_urls }
 
     async fn utxos_into_bch_unspents(&self, utxos: Vec<UnspentInfo>) -> UtxoRpcResult<BchUnspents> {
+        let to_verbose: HashSet<H256Json> = utxos
```

Comment:
Was it necessary to refactor BchCoin in this PR? My initial request was to use batch requests in the HD wallet context. It's quite dangerous to change this code in general because it may lead to SLP tokens burning.

Reply:
`ListUtxoOps::get_unspent_ordered_list` and `ListUtxoOps::get_unspent_ordered_map` have similar implementation but still they are different. So supporting of two versions of the methods can be difficult for future features.
I think that it's better to have only `ListUtxoOps::get_unspent_ordered_map` implementation, where methods like `ListUtxoOps::get_unspent_ordered_list` or `ListUtxoOps::get_mature_unspent_ordered_list` just use batch requests with only one address.
I tried not to change `BchCoin::utxos_into_bch_unspents`, `BchCoin::bch_unspents_for_spend` significantly to avoid errors.

Reply:
> I think that it's better to have only ListUtxoOps::get_unspent_ordered_map implementation, where methods like ListUtxoOps::get_unspent_ordered_list and ListUtxoOps::get_mature_unspent_ordered_list just use batch requests with only one address.

It's better if it doesn't touch the code that shouldn't have been changed in this PR (at least I assumed so).

> So supporting of two versions can be difficult.

Why will it be difficult?

Reply:
> It's better if it doesn't touch the code that shouldn't have been changed in this PR (at least I assumed so).

I refactored `BchCoin::utxos_into_bch_unspents` because `UtxoCommonOps::get_verbose_transaction_from_cache_or_rpc` was replaced with `UtxoCommonOps::get_verbose_transactions_from_cache_or_rpc` that is much more efficient because it requests verbose transactions in a batch.
We can have both `get_verbose_transaction_from_cache_or_rpc` and `get_verbose_transactions_from_cache_or_rpc`, but do we need the slow one if it's not very difficult to integrate `get_verbose_transactions_from_cache_or_rpc` into `BchCoin::utxos_into_bch_unspents`?

Reply:
> Why will it be difficult?

I thought so, before I separated the common code (like `identify_mature_unspents`) that can be easily reused in `ListUtxoOps::get_mature_unspent_ordered_list`, `ListUtxoOps::get_all_unspent_ordered_list` and `ListUtxoOps::get_unspent_lordered_ist`, so these functions shouldn't duplicate a lot of code.
Will do it.